x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/async/AsyncResponse.java

```diff
@@ -0,0 +1,17 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.core.async;
+
+import org.elasticsearch.common.io.stream.Writeable;
+
+public interface AsyncResponse extends Writeable {
```

Comment:
Indeed, we can reuse this single line toXContent but since the expirationTimeMillis is serialized as the last element of AsyncSearchResponse it will create very awkward situation with everything else. It will also limit reuse of the services. Right now AsyncResponse is a very clear abstraction - it's a expirable writable. Nothing else, nothing more.

Reply:
I see, maybe we could leave the transport serialization to the implementers but still share more? Why would sharing more code limit reuse of services?

Reply:
>  Why would sharing more code limit reuse of services?

With this being an interface, we can easily store any response. If we make it an abstract class and we will not be able to store any response that doesn't inherit from ActionResponse directly. For example, if we ever want to make any ReplicaitonResponse, BroadcastResponse or BaseNodeResponse derived response async we will have issues.

Reply:
I am not sure that we should model this so broadly and solve problems that we may never encounter. Ideally, we would avoid having an interface with only two implementers that do exactly the same thing, that is the main reason why I would go for an asbtract class and share what makes sense today, unless we already envision things to be different soon.